syntax = "proto3";

option java_package = "ex.grpc";

package myserver;

// Defines the service
service MqttServer {
    // Function invoked to send the request
    rpc connectionRequest (ConnectRequest) returns (ConnectReply) {}

    rpc disconnectionRequest (DisconnectRequest) returns (DisconnectReply) {}

    rpc publishRequest (PublishRequest) returns (PublishReply) {}

    rpc subscribeRequest (SubscribeRequest) returns (SubscribeReply) {}

    rpc showBrokersRequest (ShowBrokersRequest) returns (ShowBrokersReply) {}

    rpc showClientsRequest (ShowClientsRequest) returns (ShowClientsReply) {}
}

// The request message containing requested numbers
message ConnectRequest {
    
    string adrs=1;
    string clientID=2;
}

// The response message containing response
message ConnectReply {
   
    string response=1;
}

message DisconnectRequest {
    string adrs=1;
    string clientID=2;
}
message DisconnectReply {
    string response=1;
}

message PublishRequest {
    string adrs = 1;
    string topic = 2;
    string message = 3;
    string clientID = 4;
    
}
message PublishReply {
    string response=1;
}

message SubscribeRequest {
    string adrs = 1;
    string topic = 2;
    string clientID = 3;
    
}
message SubscribeReply {
    string response=1;
}

message ShowBrokersRequest {
    string req = 1;
}
message ShowBrokersReply {
    string response = 1;
}

message ShowClientsRequest {
    string req = 1;
}

message ShowClientsReply {
    string response = 1;
}

